backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/DatabaseDocumentNumberRecyclingService.java

```diff
@@ -75,45 +77,110 @@ public Optional<String> addForRecycling(
       }
-      return Optional.of(deletedDocumentationUnitDTO.getDocumentNumber());
+
+      assertPatternIsValid(
+          documentationOfficeAbbreviation, deletedDocumentationUnitDTO.getDocumentNumber());
+
+      return deletedDocumentationUnitDTO.getDocumentNumber();
 
     } catch (Exception e) {
       optionalDeletedDocumentationUnitDTO.ifPresent(
           deletedDocumentationUnitDTO ->
               repository.deleteById(deletedDocumentationUnitDTO.getDocumentNumber()));
-      return Optional.empty();
+      throw e;
+    }
+  }
+
+  /**
+   * Documentation unit number can be only reused if it has never been handed over or migrated from
+   * jdv.
+   *
+   * @param documentationUnitId Uuid of the reused documentation unit
+   * @throws DocumentNumberRecyclingException will throw error if has been ever used.
+   */
+  @Override
+  public void assertDocumentationUnitHasNeverBeenHandedOverOrMigrated(UUID documentationUnitId)
+      throws DocumentNumberRecyclingException {
+    if (!handoverService
+        .getEventLog(documentationUnitId, HandoverEntityType.DOCUMENTATION_UNIT)
+        .isEmpty()) {
+      throw new DocumentNumberRecyclingException(
+          "This Documentation unit has already been handed over or migrated and therefore cannot be reused");
+    }
+  }
+
+  /**
+   * Validates that the documentationUnitNumber to be recycled meets the latest condition set in
+   * documentNumberPatternConfig, that are driven from the yml format
+   *
+   * @param documentationOfficeAbbreviation office abbreviation
+   * @param documentationUnitNumber documentation unit number
+   * @throws DocumentNumberPatternException in case a pattern does not correspond to latest pattern.
+   */
+  @Override
+  public void assertPatternIsValid(
+      String documentationOfficeAbbreviation, String documentationUnitNumber)
+      throws DocumentNumberPatternException {
+    if (!documentNumberPatternConfig.hasValidPattern(
+        documentationOfficeAbbreviation, documentationUnitNumber)) {
+      throw new DocumentNumberPatternException("Pattern is invalid");
+    }
+  }
+
+  /**
+   * Validates that a documentation unit number has never been published or handed over by its
+   * status
+   *
+   * @param documentationUnitDTO dto object
+   * @throws DocumentNumberRecyclingException will through error if published / handed over or
+   *     status history not empty.
+   */
+  public void assertStatusHasNeverBeenPublished(DocumentationUnitDTO documentationUnitDTO)
+      throws DocumentNumberRecyclingException {
+    PublicationStatus status = documentationUnitDTO.getStatus().getPublicationStatus();
+    if (documentationUnitDTO.getStatusHistory().size() != 1
+        || !(status.equals(PublicationStatus.UNPUBLISHED)
+            || status.equals(PublicationStatus.EXTERNAL_HANDOVER_PENDING))) {
+      throw new DocumentNumberRecyclingException(
+          "Status has been changed once or neither unpublished nor handover pending");
```

Comment:
Hey @Nadav-B - Do you think we would safely remove this check? From my understanding, status history does not tell us anything about potential sync problems with jDV because of document number recycling. It's not harmful either, but to prevent misunderstandings and keep the amount of code to a minimum, I lean towards deletion

Reply:
I can't make up a scenario where the check for an empty event log would not be sufficient 🤔. And yes, we have been saving the original XML of each decision since day 1 and this is included into the event log. But maybe it's worth getting a third opinion on this?

Limiting the recycling of document numbers for the current year is a good idea for optimization, but since you check when the actual recycling happens, I don't think it's worth the effort.

Reply:
@rvp-c Long story short: we used the status history to determine rather we can recycle a document number. Hanna had a good point to rely only on the event log (migration and handover), meaning as long as it was never published or migrated to jdv we can reuse it. 

This check is now added on top. However, another point is to remove the check in the `getLatestStauts` and `statusHistory` for unpublished, as it might not be needed. Do you see any problem with it or shall we adapt @HPrinz idea and rely on the event log only?

Reply:
@HPrinz Ralph's feedback has no objection to your idea. However as it is not effecting the code, I will put the ticket, and if it is required we extract this to a new ticket.